src/providers/twitch/TwitchChannel.cpp

```diff
@@ -927,6 +927,69 @@ void TwitchChannel::refreshCheerEmotes()
         .execute();
 }
 
+void TwitchChannel::createClip()
+{
+    if (!this->isLive())
+    {
+        this->addMessage(makeSystemMessage(
+            "Cannot create clip while the channel is offline!"));
+        return;
+    }
+
+    this->addMessage(makeSystemMessage("Creating clip..."));
+
+    getHelix()->createClip(
+        this->roomId(),
+        [this](const HelixClip &clip) {
+            MessageBuilder builder;
+            builder.message().flags.set(MessageFlag::System);
+
+            builder.emplace<TimestampElement>();
+            // text
+            builder.emplace<TextElement>("Clip created!",
+                                         MessageElementFlag::Text,
+                                         MessageColor::System);
+            // clip link
+            builder
+                .emplace<TextElement>("Copy link to clipboard",
+                                      MessageElementFlag::Text,
+                                      MessageColor::Link)
+                ->setLink(Link(Link::CopyToClipboard,
+                               "https://clips.twitch.tv/" + clip.id));
+            // separator text
+            builder.emplace<TextElement>(" or ", MessageElementFlag::Text,
+                                         MessageColor::System);
+            // edit link
+            builder
+                .emplace<TextElement>("edit it in browser.",
+                                      MessageElementFlag::Text,
+                                      MessageColor::Link)
+                ->setLink(Link(Link::Url, clip.editUrl));
+            this->addMessage(builder.release());
+        },
+        [this] {
+            const QString failureText =
+                "Failed to create a clip. Since this feature is new "
+                "and it requires extra scopes";
```

Comment:
```suggestion
                "Failed to create a clip. Since this feature is new";
```
I'm not 100% on declaring 'new' in the error, but the scopes part is irrelevant to the average user.

Reply:
I guess that's fair, makes the error message shorter which I think is good because I also realized I have to include something about streamer having clips disabled, so I'll just edit that myself. Thanks for the input :+1: